src/array/dyn_geometry_array.rs

```diff
@@ -0,0 +1,210 @@
+use crate::array::zip_validity::ZipValidity;
+use crate::array::{
+    GeometryCollectionArray, LineStringArray, MixedGeometryArray, MultiLineStringArray,
+    MultiPointArray, MultiPolygonArray, PointArray, PolygonArray, RectArray, WKBArray,
+};
+use crate::datatypes::GeoDataType;
+use crate::trait_::GeoArrayAccessor;
+use crate::GeometryArrayTrait;
+use arrow_array::OffsetSizeTrait;
+use arrow_buffer::bit_iterator::BitIterator;
+
+impl dyn GeometryArrayTrait {
+    pub fn value_as_geo(&self, i: usize) -> geo::Geometry {
```

Comment:
I'll link to https://github.com/geoarrow/geoarrow-rs/pull/291#discussion_r1423094084. I don't think we'd ever want to incur the downcasting cost on every row